Solutions/Salesforce Service Cloud/Analytic Rules/Salesforce-BruteForce.yaml

```diff
@@ -0,0 +1,42 @@
+id: 5a6ce089-e756-40fb-b022-c8e8864a973a
+name: Brute force attack against user credentials
+description: |
+  'Identifies evidence of brute force activity against a user based on multiple authentication failures 
+  and at least one successful authentication within a given time window. Note that the query does not enforce any sequence,
+  and does not require the successful authentication to occur last.
+  The default failure threshold is 10, success threshold is 1, and the default time window is 20 minutes.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: SalesforceServiceCloud
+    dataTypes:
+      - SalesforceServiceCloud
+queryFrequency: 20m
+queryPeriod: 20m
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - CredentialAccess
+relevantTechniques:
+  - T1110
+query: |
+  let failureCountThreshold = 10;
+  let successCountThreshold = 1;
+  SalesforceServiceCloud
+  | where EventType == "Login"
+  | summarize 
+        StartTime = min(TimeGenerated), 
+        EndTime = max(TimeGenerated), 
+        IpAddresses = make_set (ClientIp, 100),
+        FailureCount = countif(LoginStatus !='LOGIN_NO_ERROR'),
+        SuccessCount = countif(LoginStatus =='LOGIN_NO_ERROR')
+    by User, UserId, UserType 
+    | where FailureCount >= failureCountThreshold and SuccessCount >= successCountThreshold
```

Comment:
I would recommend that you include a check that the Failure is before the Success, otherwise you will get hits for Success then Failure which can happen for a variety of reasons.

Reply:
Query updated with check for failure

Reply:
@rushriva - You should remove the information in the description that states it does not care about sequence as it does now - `Note that the query does not enforce any sequence,
  and does not require the successful authentication to occur last.`

Reply:
done